### PR TITLE
Add delete_study api for storage package

### PR DESCRIPTION
**Pull request by c-bata**
close https://github.com/pfnet/optuna/issues/421

* [x] Add storage api to delete study
* [x] Add `delete-study` subcommand on CLI

### CODE REVIEW

File: optuna/cli.py [hunk 1]
```diff
@@ -82,6 +82,27 @@ def take_action(self, parsed_args):
         print(study_name)
 
 
+class DeleteStudy(BaseCommand):
```
Comment (c-bata) [inline optuna/cli.py]:
```
 $ optuna help
usage: optuna [--version] [-v | -q] [--log-file LOG_FILE] [-h] [--debug]
              [--config CONFIG] [--storage STORAGE]

optional arguments:
  --version            show program's version number and exit
  -v, --verbose        Increase verbosity of output. Can be repeated.
  -q, --quiet          Suppress output except warnings and errors.
  --log-file LOG_FILE  Specify a file to log output. Disabled by default.
  -h, --help           Show help message and exit.
  --debug              Show tracebacks on errors.
  --config CONFIG      Config file path. (default=$HOME/.optuna.yml)
  --storage STORAGE    DB URL. (e.g. sqlite:///example.db)

Commands:
  complete       print bash completion command (cliff)
  create-study   
  dashboard      
  delete-study   
  help           print detailed help for another command (cliff)
  storage upgrade  
  studies        Command base class for providing a list of data as output.
  study optimize  
  study set-user-attr  
(venv) $ optuna help delete-study
usage: optuna delete-study [-h] [--study-name STUDY_NAME]

optional arguments:
  -h, --help            show this help message and exit
  --study-name STUDY_NAME
                        A human-readable name of a study to distinguish it
                        from others.
```

**Comment by c-bata** [comment]
```
(venv) $ optuna create-study --study-name delete-test --storage sqlite:///db.sqlite3
[I 2019-09-14 14:27:54,250] A new study created with name: delete-test
delete-test
(venv) $ python examples/sklearn_simple.py  # load_study(study_name='delete-test', storage='sqlite:///db.sqlite3')
...
(venv) $ sqlite3 db.sqlite3 
SQLite version 3.29.0 2019-07-10 17:32:03
Enter ".help" for usage hints.
sqlite> .header on
sqlite> .mode columns
sqlite> select * from studies;
study_id    study_name   direction 
----------  -----------  ----------
1           delete-test  MINIMIZE  
sqlite> select count(*) from trials;
count(*)  
----------
20        
sqlite>  quit
```

```
(venv) $ optuna delete-study --study-name delete-test --storage sqlite:///db.sqlite3
(venv) $ sqlite3 db.sqlite3 
SQLite version 3.29.0 2019-07-10 17:32:03
Enter ".help" for usage hints.
sqlite> .mode column
sqlite> .header on
sqlite> select * from studies;
sqlite> select * from trials;
sqlite> 
```

File: optuna/storages/in_memory.py [hunk 2]
```diff
@@ -54,6 +54,20 @@ def create_new_study(self, study_name=None):
...
+        self._check_study_id(study_id)
+
+        with self._lock:
+            self.trials = []
+            self.param_distribution = {}
+            self.direction = structs.StudyDirection.NOT_SET
+            self.study_user_attrs = {}
+            self.study_system_attrs = {}
+            self.study_name = DEFAULT_STUDY_NAME_PREFIX + IN_MEMORY_STORAGE_STUDY_UUID
+        return
```
Comment (sile) [inline optuna/storages/in_memory.py]: I think this `return` can be removed.
```suggestion
```

File: tests/storages_tests/test_storages.py [hunk 3]
```diff
@@ -139,6 +139,22 @@ def test_create_new_study_with_name(storage_mode, cache_mode):
         assert study_name == storage.get_study_name_from_id(study_id)
```
Comment (sile) [inline tests/storages_tests/test_storages.py]: Please add a test like the following scenario for RDBStorage:
1. Create multiple studies
2. Remove one of them
3. Assert the deleted study was actually removed from the study list

File: optuna/storages/rdb/storage.py [hunk 4]
```diff
@@ -98,6 +98,15 @@ def create_new_study(self, study_name=None):
 
         return study.study_id
 
+    def delete_study(self, study_id):
+        # type: (int) -> None
```
Comment (toshihikoyanase) [inline optuna/storages/rdb/storage.py]: Please add a blank line after the type hint.

Comment (c-bata) [reply optuna/storages/rdb/storage.py]: Thank you! I fixed.

**Pull request merged**